lib/hammer_cli_katello/content_view_version.rb

```diff
@@ -385,6 +385,130 @@ def export_json(content_view_version, repositories)
       end
     end
 
+    class ImportCommand < HammerCLIForeman::Command
+      include HammerCLIForemanTasks::Async
+      include HammerCLI::Messages
+      include HammerCLIKatello::LocalHelper
+      include HammerCLIKatello::ApipieHelper
+
+      attr_accessor :export_tar_dir, :export_tar_file, :export_tar_prefix
+
+      desc "Import a content view version"
+
+      command_name "import"
+
+      success_message _("Content view imported.")
+      failure_message _("Could not import the content view")
+
+      option(
+        '--export-tar',
+        'EXPORT_TAR',
+        _("Location of export tar on disk.")
+      )
+
+      validate_options do
+        option(:option_export_tar).required
+      end
+
+      build_options
+
+      def execute
+        unless File.exist?(options['option_export_tar'])
+          raise "Export tar #{options['option_export_tar']} does not exist"
+        end
+
+        self.export_tar_file = File.basename(options['option_export_tar'])
+        self.export_tar_prefix = @export_tar_file.gsub('.tar', '')
+        self.export_tar_dir = File.dirname(options['option_export_tar'])
+        untar_export
+
+        export_json = read_json
+
+        cv = content_view(export_json['name'])
+        sync_repositories(export_json['repositories'])
+
+        publish(
+          cv['id'],
+          export_json['major'],
+          export_json['minor'],
+          repos_units(export_json['repositories'])
+        )
+        return HammerCLI::EX_OK
+      end
+
+      def untar_export
+        Dir.chdir(@export_tar_dir) do
+          `tar -xf #{@export_tar_file}`
+        end
+
+        Dir.chdir("#{@export_tar_dir}/#{@export_tar_prefix}") do
+          if File.exist?(@export_tar_file.gsub('.tar', '-repos.tar'))
+            `tar -xf #{@export_tar_file.gsub('.tar', '-repos.tar')}`
+          else
+            raise "Export repos tar file is missing"
+          end
+        end
+      end
+
+      def read_json
+        json_file = @export_tar_file.gsub('tar', 'json')
+        json_file = "#{@export_tar_dir}/#{@export_tar_prefix}/#{json_file}"
+        json_file = File.read(json_file)
+        JSON.parse(json_file)
+      end
+
+      def sync_repositories(repositories)
+        repositories.each do |repo|
+          library_repo = index(
+            :repositories,
+            'label' => repo['label'],
+            'library' => true
+          )
+          library_repo = library_repo.first
```

Comment:
Searching by label just returns everything, the label is ignored.

here's how I worked around it (might not be the most ruby way)

```ruby
          library_repos = index(
            :repositories,
            'content_view_version_id' => 1
          )

          library_repo = library_repos.select do |candidate_repo|
            candidate_repo['label'] == repo['label']
          end

          library_repo = library_repo.first

```